src/ignore.c

```diff
@@ -74,6 +112,8 @@ static int does_negate_rule(int *out, git_vector *rules, git_attr_fnmatch *match
 			error = 0;
 			goto out;
 		}
+
+		error = 0;
```

Comment:
Sorry, in reading through this it's not clear to me when this would need to be cleared.  This is probably just something that I'm overloooking?